assignment3/Dictionary.java

```diff
@@ -1,35 +1,50 @@
 package com.katermar.assignment3;
 
+import java.util.Arrays;
 import java.util.HashSet;
 
 /**
  * Created by katermar on 6/8/2017.
  */
 public class Dictionary {
-    private HashSet<String> dictionary = new HashSet<>();
-    private HashSet<String> prefixes = new HashSet<>();
+    private final HashSet<String> dictionary;
+    private final HashSet<String> prefixes;
 
-    public boolean isPrefix(String prefix){
-      return prefixes.contains(prefix);
+    public Dictionary(HashSet<String> dictionary, HashSet<String> prefixes) {
+        this.dictionary = dictionary;
+        this.prefixes = prefixes;
     }
 
-    public boolean isWord(String word) {
-        return dictionary.contains(word);
+    public Dictionary(String[] dictionary, String[] prefixes) {
+        this.dictionary = new HashSet<>(Arrays.asList(dictionary));
+        this.prefixes = new HashSet<>(Arrays.asList(prefixes));
     }
 
-    public HashSet<String> getDictionary() {
-        return dictionary;
+    public Dictionary(String[] dictionary) {
+        this.dictionary = new HashSet<>(Arrays.asList(dictionary));

```

Comment:
There is some code duplication between this constructor and the one accepting HashSet. You could just use constructor chaining here:
```
public Dictionary(String[] dictionary) {
  super(new HashSet<>(Arrays.asList(dictionary));
}
```